### ISSUE TITLE
Default branch build gaps after throttling change

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#916]
Analysis deletes versions on changes and we lose the associated builds. Since we now delay default branch builds until the next day, we've created a gap in the build matrix and show "?" until the builds kick in.

I believe we've got the following options to address this

1. Roll back change to build trigger
2. Introduce additional tracking fields to support latest version display on package page during cool-down period and keep version with builds
3. Introduce a new “intraday latest branch” version for the purpose of the package page display that is without builds but won’t get built until the cool-down period has expired. Then roll over once that build has happened.
4. Roll back build throttling and introduce a longer version update window during analysis for branch changes

I think there are only two non-hacky ways to do this:

- Option 1 - roll back and pay the build price
- Option 4 - roll back build throttle + longer version update window

I’m leaning towards 4. The way this would work is that during analysis we ignore changes to default branch versions within X hours, say 8 or 12. That way, any changes that happen within that window get collated into a single refresh, such that we’d build at most 2 or 3 times per day while also not falling too far behind with the “last updated” display on the website. It shouldn’t really matter if the latest change (for just the default branch) is 8h or 12h behind.

We'd pick up a few more builds than we do with current throttling (unless we decide to make the window 24h and accept the update metadata on the package page to be more outdated).

The upside is that the throttling query becomes simpler again, albeit at the expense of more complex analysis version handling. OTOH, that's outside of an SQL query and probably easier to test and understand.

**Comment by daveverwer** [comment]
Some thoughts:

* I think we *have* to address this in some way. Having so many "?" icons really reduces the effectiveness of the build system.
* I can see why you're reluctant to introduce new fields, but I think option 2 would probably be the best experience for visitors if the complexity didn't exist.
* I also think option 4 is the best, given the complexity of option 2.

I think there may be a change we can make that will make option 4 even more acceptable. If we changed the accuracy of the "X hours/days ago" descriptions next to versions:

<img width="264" alt="Screen Shot 2021-02-04 at 11 43 37@2x" src="https://user-images.githubusercontent.com/5180/106888279-3d2c0500-66de-11eb-87a3-f502e61b5c9b.png">

If we changed that to "Modified within the last day" that's still great information as those are supposed to let you know if the package is receiving attention or is abandoned and it would mean that the data is still accurate.

The big downside of this change is that the "Recent Releases" list on the home page would be less useful as they'd all say "Released within the last day". We have two choices there. We could keep "X hours ago" for *releases* and "within the last day" for branches, but that feels weird. Or, we could remove the "time ago" from the list on the home page. They'd still be listed in chronological order, but just not labelled with a specific time.
